File: app/src/main/java/com/prolocity/patchtracker/data/Session.kt
package com.prolocity.patchtracker.data

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.time.LocalDate

@Entity(tableName = "sessions")
data class Session(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val name: String,
    val createdDate: LocalDate,
    val isCurrent: Boolean = false,
    // Set once the session has been exported to a backup. A finalized session's patch awards
    // can no longer be added to/edited, and only a finalized, non-current session can be deleted.
    val isFinalized: Boolean = false
)

File: ios/PatchTracker/Data/Models/Session.swift
import Foundation
import SwiftData

/// A round of league play. The app always has exactly one `isCurrent` session, and new patch
/// awards default to it. `isFinalized` is set once the session has been exported to a backup —
/// a finalized session's awards are locked, and only a finalized, non-current session can be
/// deleted.
@Model
final class Session {
    var name: String
    var createdDate: Date
    var isCurrent: Bool
    var isFinalized: Bool

    @Relationship(deleteRule: .cascade, inverse: \PatchAwardEvent.session)
    var events: [PatchAwardEvent] = []

    init(name: String, createdDate: Date, isCurrent: Bool = false, isFinalized: Bool = false) {
        self.name = name
        self.createdDate = createdDate
        self.isCurrent = isCurrent
        self.isFinalized = isFinalized
    }
}
